daemon/CMakeLists.txt

```diff
@@ -46,15 +46,27 @@ if(WITH_BOOTSTRAP)
   endif()
 endif()
 
+# cmake interface library for bunch of cmake hacks to fix final link order
+add_library(hax_and_shims_for_cmake INTERFACE)
+if(WIN32)
+  target_link_libraries(hax_and_shims_for_cmake INTERFACE uvw oxenmq::oxenmq -lws2_32 -lshlwapi -ldbghelp -luser32 -liphlpapi -lpsapi -luserenv)
+endif()
```

Comment:
This seems counterproductive; why not just link to all of these directly in the one single place that links to this interface?

Reply:
link order matters here

Reply:
Doesn't answer my question as to why inject a single-use INTERFACE library here.

Reply:
that is exactly what this is doing.

Reply:
it needs to do that. but in a very specific order

Reply:
You link to this interface in exactly one place, which is going to be equivalent to just linking to the same list of items rather than the interface at the point where you are linking to the interface.

Reply:
oxemq target link order was borked and i did not want to figure out how to fix it in an upstreamable way, i used this approach

Reply:
"I put in a dirty hack because I couldn't be bothered to fix it properly" does not make me happy.

Reply:
it also does not make me happy, when deadlines are imposed these kinds of hacks becomes the norm. this is a consistent source of our cruft across the repos, do it right or do it twice.